tmail-backend/blob/blobid-list/src/main/java/com/linagora/tmail/blob/blobid/list/SingleSaveBlobStoreDAO.java

```diff
@@ -25,17 +25,16 @@ public SingleSaveBlobStoreDAO(BlobStoreDAO blobStoreDAO,
 
     public Mono<Void> save(BucketName bucketName, BlobId blobId, byte[] data) {
         return Mono.from(blobIdList.isStored(blobId))
-                .filter(isStored -> isStored.equals(false))
-                .switchIfEmpty(Mono.empty())
-                .then(Mono.from(blobIdList.store(blobId))
-                        .then(Mono.from(blobStoreDAO.save(bucketName, blobId, data)))
-                        .onErrorResume(error -> Mono.from(blobIdList.remove(blobId)).then()))
+                .filter(isStored -> isStored.equals(true))
```

Comment:
Why this false turned into true?

Reply:
old: filter `false` -> Do nothing if empty
new: filter `true` -> Save blob if empty
=> same logic

Reply:
Ok I find it very tricky to understand.

How about

```
Mono.from(blobIdList.isStored(blobId))
    .flatMap(isStored -> {
        if(isStored) {
            return Mono.empty();
        }
        return blobStoreDAO.save(bucketName, blobId, data))
                                .then(Mono.from(blobIdList.store(blobId)));
   })
```

Reply:
(y) It is very clear. I'm just obsessed with "Don't use if in FP"